notebook/notebook/notebook.py

```diff
@@ -88,72 +96,80 @@ def web_authenticated_workshop_guest_only(fun):
         f'--NotebookApp.base_url={service_base_path}/instance/{notebook_token}/',
         "--ip", "0.0.0.0", "--no-browser", "--allow-root"
     ]
-    if 'workshop_image' in userdata:
-        image = userdata['workshop_image']
-    else:
-        image = DEFAULT_WORKER_IMAGE
-    volumes = [
-        kube.client.V1Volume(
-            name='deploy-config',
-            secret=kube.client.V1SecretVolumeSource(
-                secret_name='deploy-config'))
-    ]
-    volume_mounts = [
-        kube.client.V1VolumeMount(
-            mount_path='/deploy-config',
-            name='deploy-config',
-            read_only=True)
-    ]
 
-    user_id = userdata['id']
-    ksa_name = userdata.get('ksa_name')
+    if service == 'notebook':
+        service_account_name = userdata.get('ksa_name')
 
-    bucket = userdata.get('bucket_name')
-    if bucket is not None:
+        bucket = userdata['bucket_name']
```

Comment:
If we do this we open up to exception raised from missing key; will this lead to a 500 like in Flask? We probably want to return a 401 or 403?

Reply:
This is what we did before: https://github.com/hail-is/hail/pull/7112/files#diff-ceb9d5f72e3872f06586c8b39ed0679bL47

Bucket is required.  The reason it is a get in the diff is because the old code handled handled both the notebook and workshop paths, now it only handles the notebook path, here the bucket is required for all users.

Reply:
Ah right, I misread this as being in the decorator. Yep.